cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp

```diff
@@ -654,16 +654,26 @@ std::shared_ptr<Analyzer::Expr> Substrait2AnalyzerExprConverter::buildInValuesEx
 std::shared_ptr<Analyzer::Expr> Substrait2AnalyzerExprConverter::buildExtractExpr(
     const substrait::Expression_ScalarFunction& s_scalar_function,
     const std::unordered_map<int, std::string> function_map,
+    const ExtractField& extract_field,
     std::shared_ptr<std::unordered_map<int, std::shared_ptr<Analyzer::Expr>>>
         expr_map_ptr) {
-  CHECK(s_scalar_function.arguments_size() == 2);
-  CHECK(s_scalar_function.arguments(0).has_enum_());
-  std::string time_unit = s_scalar_function.arguments(0).enum_().specified();
-  CHECK(time_unit != *&::google::protobuf::internal::GetEmptyStringAlreadyInited());
-  CHECK(s_scalar_function.arguments(1).has_value());
-  auto from_expr =
-      toAnalyzerExpr(s_scalar_function.arguments(1).value(), function_map, expr_map_ptr);
-  return ExtractExpr::generate(from_expr, time_unit);
+  if (extract_field == ExtractField::NONE) {
+    CHECK(s_scalar_function.arguments_size() == 2);
+    CHECK(s_scalar_function.arguments(1).has_value());
+    auto from_expr = toAnalyzerExpr(
+        s_scalar_function.arguments(1).value(), function_map, expr_map_ptr);
+    CHECK(s_scalar_function.arguments(0).has_enum_());
+    std::string time_unit = s_scalar_function.arguments(0).enum_().specified();
+    CHECK(time_unit != *&::google::protobuf::internal::GetEmptyStringAlreadyInited());
```

Comment:
emmm can igonre `*&`? seems hard to understand.

Reply:
updated